src/tool_operate.c

```diff
@@ -653,8 +653,12 @@ static CURLcode post_per_transfer(struct GlobalConfig *global,
       errorf(config->global, "curl: (%d) Failed writing body", result);
     }
     if(result && config->rm_partial) {
-      notef(global, "Removing output file: %s", outs->filename);
-      unlink(outs->filename);
+      struct_stat st;
+      stat(outs->filename, &st);
+      if(S_ISREG(st.st_mode)) {
+        notef(global, "Removing output file: %s", outs->filename);
+        unlink(outs->filename);
```

Comment:
A failure here should at least show a warning, if not return an error code from the program. And possibly a notef if it does successfully delete the file.

Reply:
In this case, it already has an error and that is why this function is executed. It would be a bit strange to change the error code I think. Right?

Reply:
If it's going to return an error anyway, even in the absence of --fail, then that seems fine.